Assets/MRTK/SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs

```diff
@@ -452,77 +463,92 @@ public override int GetHashCode()
 
         #region IMixedRealitySourcePoseHandler Implementation
 
+        private static readonly ProfilerMarker OnSourceLostPerfMarker = new ProfilerMarker("[MRTK] BaseControllerPointer.OnSourceLost");
+
         /// <inheritdoc />
         public override void OnSourceLost(SourceStateEventData eventData)
         {
-            base.OnSourceLost(eventData);
-
-            if (eventData.SourceId == InputSourceParent.SourceId)
+            using (OnSourceLostPerfMarker.Auto())
             {
-                if (requiresHoldAction)
-                {
-                    IsHoldPressed = false;
-                }
+                base.OnSourceLost(eventData);
 
-                if (IsSelectPressed)
+                if (eventData.SourceId == InputSourceParent.SourceId)
                 {
-                    CoreServices.InputSystem.RaisePointerUp(this, pointerAction, Handedness);
-                }
+                    if (requiresHoldAction)
+                    {
+                        IsHoldPressed = false;
+                    }
+
+                    if (IsSelectPressed)
+                    {
+                        CoreServices.InputSystem.RaisePointerUp(this, pointerAction, Handedness);
+                    }
 
-                IsSelectPressed = false;
+                    IsSelectPressed = false;
+                }
             }
         }
 
         #endregion IMixedRealitySourcePoseHandler Implementation
 
         #region IMixedRealityInputHandler Implementation
 
+        private static readonly ProfilerMarker OnInputUpPerfMarker = new ProfilerMarker("[MRTK] BaseControllerPointer.OnInputUp");
+
         /// <inheritdoc />
         public override void OnInputUp(InputEventData eventData)
         {
-            if(!IsInteractionEnabled) { return; }
+            using (OnInputUpPerfMarker.Auto())
```

Comment:
Similar Q in this file around IsInteractionEnabled checks (i.e. is it worth doing it if it's a no-op)